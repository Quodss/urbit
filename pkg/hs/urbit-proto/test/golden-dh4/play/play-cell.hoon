:*  {@ @}
    {a=@ @}
    {a=$ a}
    {a={@ $} +3.a}
    {a=$-(@ $) $-(b=@ (a b))}
==
