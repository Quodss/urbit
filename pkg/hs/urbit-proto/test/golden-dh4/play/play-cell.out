PARSED

:*  {@ @}
    {a=@ @}
    {a=$ a}
    {a={@ $} +3.a}
    {a=$-(@ $) $-(b=@ (a b))}
==

SOFT

:*  {@ @}
    {a=@ @}
    {a=$ a}
    {a={@ $} +3.a}
    {a=$-(@ $) $-(b=@ (a b))}
==

HARD

:*  {@ @}
    {a=@ @}
    {a=$ a}
    {a={@ $} +3.a}
    $:  ^=  a
        $?  +7
        $|  [$;(@ *) *]
        ++  %  $
        --
        $?  +7
        $|  [$;(b=@ *) *]
        ++  %
          =+  a
          %.+2(+6 b:+3)
        --
    ==
==

CODE

:*  {@ @}
    {a=@ @}
    {a=$ +2}
    {a={@ $} +5}
    {a=$-(@ $) $-(b=@ (+6 +4))}
==

TYPE

[$ $ $ $ $]

BASE

:*  {@ @}
    {a=@ @}
    {a=$ +2}
    {a={@ $} +5}
    {a=$-(@ $) $-(b=@ %.+6(+6 +4))}
==